addons/forward/models.py

```diff
@@ -33,6 +35,17 @@ def after_register(self, node, registration, user, save=True):
 
         return clone, None
 
+    def save(self, request=None, *args, **kwargs):
+        super(NodeSettings, self).save(*args, **kwargs)
+        if request:
+            if not hasattr(request, 'user'):  # TODO: remove when Flask is removed
+                _, user_id = get_request_and_user_id()
+                user = OSFUser.load(user_id)
+            else:
+                user = request.user
+
+            self.owner.check_spam(user, {'addons_forward_node_settings__url'}, get_headers_from_request(request))
```

Comment:
wondering if we should only check spam if the url has actually been changed - instead of every time the forward addon is saved.

Reply:
Hmmm. From a performance standpoint, I doubt it matters much, since I don't think the forward addon is used much. If we trust the spam filter a lot, then it's better to check each time, in case we modify the spam filter to have a better understanding of what is spam. If we are concerned about false positives (which I don't believe we are), then not checking as frequently is better.

Reply:
It's a bit of a trade off, if we only want to run this when the user changes something it should probably go back in the serializer.

Reply:
I agree with @brianjgeiger, i'm not super concerned about false positives and this isn't going to be run a lot.  I do prefer having it on the model since it's being called in 3 places -

Reply:
👍 agreed  I think this is better from a data integrity standpoint and also easier to remove if we depreciate the Forward addon.